drivers/spi/spi_nrfx_spim.c

```diff
@@ -759,6 +759,12 @@ static int spi_nrfx_init(const struct device *dev)
 #endif
 	return pm_device_driver_init(dev, spim_nrfx_pm_action);
 }
+
+static int spi_nrfx_deinit(const struct device *dev)
+{
+	return 0;
```

Comment:
I think that when CONFIG_PM_DEVICE=n this function should actually call spim_suspend(dev);

Reply:
100% agree, this PR is a bit out of date, see https://github.com/zephyrproject-rtos/zephyr/blob/ad4c3e3e9afe1385ab80adf3fa2d3309fad68c60/drivers/spi/spi_nrfx_spim.c#L769-L788